qsiprep/utils/grouping.py

```diff
@@ -991,6 +1010,26 @@ def group_for_eddy(all_dwi_fmap_groups):
                '.../mixed_fmaps/sub-1/fmap/sub-1_dir-PA_run-1_epi.nii.gz']},
       'concatenated_bids_name': 'sub-1'}]
 
+    Val's scenario
```

Comment:
I think this might be what's happening with @valeriejill's problem (#920).

Reply:
Should we keep this behavior if there are no additional uses of b0fieldidentifier or IntendedFors?

Reply:
That's a really hard question. If you _didn't_ want to have those runs merged, how would you specify that except by not including those fields?